Code/cicp_inserter/CICPInserter.cpp

```diff
@@ -60,18 +62,72 @@ namespace CICP_Inserter {
 			         chunk_type[1] == 'I' &&
 			         chunk_type[2] == 'C' &&
 			         chunk_type[3] == 'P') {
-				return std::unexpected{ GetInsertionIndexError{ GetInsertionIndexErrorCode::CICPChunkAlreadyExists, { cicp_chunk_already_exists, newline } } };
+				if (!overwrite_cicp) {
+					return std::unexpected{ GetInsertionIndexError{ GetInsertionIndexErrorCode::CICPChunkAlreadyExists, { cicp_chunk_already_exists, newline } } };
+				}
+				// TODO: Handle multiple cICP chunks
+				cicp_index = index;
 			}
 		}
 
 		if (plte_index == SIZE_MAX && idat_index == SIZE_MAX) {
 			return std::unexpected{ GetInsertionIndexError{ GetInsertionIndexErrorCode::CouldNotFindInsertionPoint, { could_not_find_insertion_point, newline } } };
 		}
 
-		// The cICP chunk must come before the PLTE and IDAT chunks.
-		// It could come anywhere before that (but after IHDR).
-		// Inserting it just before the first chunk it must proceed is fine.
-		return std::min(plte_index, idat_index);
+
+		// There are several combinations of potential splits.
+		// 
+		// A PNG might look like {IHDR, PLTE, IDAT, IEND}.
+		// In this case, we can split as {IHDR}, {PLTE, IDAT, IEND}.
+		// 
+		// After the cICP chunk is inserted, picking the first of (PLTE/IDAT)
+		// and overwriting the cICP results in the same split:
+		// {IHDR}, {PLTE, IDAT, IEND}
+		//
+		// However, what if the PNG looked like {IHDR, PLTE, cICP, IDAT, IEND}.
+		// If we split in front of the first (PLTE/IDAT) & remove the cICP, we'll see these splits:
+		// {IHDR}, {PLTE}, {IDAT, IEND}
+		//
+		// Alternatively, we could simply overwrite the cICP in-place, falling back to first of
+		// (PLTE/IDAT) if no cICP is found. Then that last PNG's splits would look like:
+		// {IHDR, PLTE}, {IDAT, IEND}
+		//
+		// This gives us a consistent before & after scenario.
```

Comment:
I guess that is better from a "create known invalid test files" point of view.

On the other hand, recovering from a situation where some other tool added `cICP`, but in the wrong place, is no longer possible.

Reply:
Yeah. This relies on the PNG being valid.
I figured the rest of the tool also relies on the PNG being valid, too.

I think a better route would be to generally not rely on that and warn/fix.
But that is probably a bigger project where we should open an issue and work on it separately. What do you think?

Reply:
Agree this is a separate issue, and needs discussion. Not for this PR